File: class18/python/AP1811-SomaAteZero.py
# SomaAteZero — Aula 18
# Soma números até o usuário digitar zero (loop enquanto com flag)

soma = 0
while True:
    n = float(input("Digite um número (0 para sair): "))
    if n == 0:
        break
    soma += n
print(f"Soma total: {soma:.2f}")

File: class18/java/AP1811-SomaAteZero.java
import java.util.Scanner;
public class SomaAteZero {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        double soma = 0;
        while (true) {
            System.out.print("Digite um número (0 para sair): ");
            double n = sc.nextDouble();
            if (n == 0) break;
            soma += n;
        }
        System.out.printf("Soma total: %.2f%n", soma);
        sc.close();
    }
}